lib/pipeline/RequestHandler.js

```diff
@@ -115,18 +123,15 @@ class RequestHandler extends Duplex {
   }
 
   _flush() {
-    var i;
-
-    //faster than deque.
-    for(i = 0; i < this.queue.length && this.reading ; i++) {
-      this.reading = this.push(this.queue[i]);
+    while (this.queue.length) {
```

Comment:
Isn't `this.reading` part required? might the write stream be closed or something like that?